### PR TITLE
Better handle invalid presets

### PR DESCRIPTION
**Pull request by henriquegemignani**
- Disable menu actions that need a preset if it's invalid
- Add a clickable link that opens the preset dir when there's an error
- Show full error stack for unknown errors

image
image

### CODE REVIEW

File: randovania/gui/lib/common_qt_lib.py [hunk 1]
```diff
@@ -237,13 +237,26 @@ def set_clipboard(text: str):
...
 
 
-def open_directory_in_explorer(path: Path):
-    if platform.system() == "Windows":
-        os.startfile(path)
-    elif platform.system() == "Darwin":
-        subprocess.run(["open", path])
-    else:
-        subprocess.run(["xdg-open", path])
+def open_directory_in_explorer(path: Path, fallback_dialog: tuple[str, str, QtWidgets.QWidget] | None = None):
```
Comment (duncathan) [inline randovania/gui/lib/common_qt_lib.py:240]: a NamedTuple for fallback dialog might be nice

File: randovania/gui/widgets/select_preset_widget.py [hunk 2]
```diff
@@ -316,17 +325,40 @@ def on_preset_changed(self, preset: VersionedPreset | None):
...
+                    f"or delete this preset.</p>"
+                    f"{exception_desc}"
                 )
 
         self.create_preset_description.setText(description)
+        self._can_generate = can_generate
         self.CanGenerate.emit(can_generate)
 
+    def _on_click_create_preset_description(self, link: str):
+        info = re.match(r"^open-preset://([0-9a-f\-]{36})$", link)
```
Comment (duncathan) [inline randovania/gui/widgets/select_preset_widget.py:348]: might as well `re.compile` this for reuse imo

**Review by duncathan — CHANGES_REQUESTED** [review]

File: randovania/gui/widgets/select_preset_widget.py (hunk 2 above)
Comment (henriquegemignani) [reply randovania/gui/widgets/select_preset_widget.py:348]: I doubt this search is done more than once per app, and most of the time none. Better performance doing it here.

**Comment by henriquegemignani** [comment]
It is, but the stack trace is pre-formatted so it doesn't line break, which causes the text above to not be either

**Review by duncathan — APPROVED** [review]

**Pull request merged**